pkg/storage/sasNode.go

```diff
@@ -215,6 +223,23 @@ func (sas *sasStorage) NodeUnpublishVolume(ctx context.Context, req *csi.NodeUnp
 		return nil, status.Errorf(codes.DataLoss, "(unpublish) filesystem seems to be corrupted: %v", err)
 	}
 
+	if !connector.Multipath {
+		// If we didn't discover the multipath device initially, double check that we didn't just miss it
+		// Detach the discovered devices if they are found
+		klog.Info("Device saved as non-multipath. Searching for additional devices before Detach")
+		if connector.IoHandler == nil {
+			connector.IoHandler = &saslib.OSioHandler{}
+		}
+		discoveredMpathName, devices := saslib.FindDiskById(klog.FromContext(ctx), wwn, connector.IoHandler)
+		if (discoveredMpathName != connector.OSPathName) && (len(devices) > 0) {
+			klog.V(3).Infof("Found additional linked devices: %s, %v", discoveredMpathName, devices)
```

Comment:
As far as useful logs, I feel like this line is more important than 226 which always gets printed. It would always be good to know when we missed a multipath device and then added it back in.

Reply:
Changed logging priority in this section